consensus/misc/rewards.go

```diff
@@ -33,6 +33,55 @@ func QuaiToQi(block *types.WorkObject, exchangeRate *big.Int, difficulty *big.In
 	return new(big.Int).Quo(qiByQuai, CalculateQuaiReward(difficulty, exchangeRate))
 }
 
+// ComputeConversionAmountInQuai computes the amount of conversion volume in
+// quai for the given set of newInboundEtxs using the values from the header
+func ComputeConversionAmountInQuai(header *types.WorkObject, newInboundEtxs types.Transactions) *big.Int {
+	conversionAmountInQuai := big.NewInt(0)
+	for _, etx := range newInboundEtxs {
+		// If the etx is conversion
+		if types.IsConversionTx(etx) {
+			value := etx.Value()
+			// If to is in Qi, convert the value into Qi
```

Comment:
If the flows are balanced over the conversion period, I don't know if we need to apply a cubic discount. We may consider netting the quai and qi conversion flows when applying the discount.